tests/rptest/tests/nodes_decommissioning_test.py

```diff
@@ -111,12 +112,25 @@ def requested_status():
 
         wait_until(requested_status, timeout_sec=timeout_sec, backoff_sec=1)
 
-    def _set_recovery_rate(self, rate):
+    def _set_recovery_rate(self, rate, timeout_sec=30):
         # use admin API to leverage the retry policy when controller returns 503
-        patch_result = Admin(self.redpanda).patch_cluster_config(
-            upsert={"raft_learner_recovery_rate": rate})
-        self.logger.debug(
-            f"setting recovery rate to {rate} result: {patch_result}")
+        def is_rate_set():
+            try:
+                patch_result = Admin(self.redpanda).patch_cluster_config(
```

Comment:
Admin wrapper already implements retries for a list of status codes, I think we can simply add 504 to this list (maybe even do it globally) and increase the number of retries if necessary.

Reply:
something like [this](https://github.com/redpanda-data/redpanda/blob/dev/tests/rptest/tests/partition_balancer_test.py#L152)